deno/lib/types.ts

```diff
@@ -3488,6 +3488,175 @@ export class ZodNaN extends ZodType<number, ZodNaNDef> {
   };
 }
 
+/////////////////////////////////////////
+/////////////////////////////////////////
+//////////                     //////////
+//////////      ZodFile         //////////
+//////////                     //////////
+/////////////////////////////////////////
+/////////////////////////////////////////
+
+type ZodFileCheck =
+  | { kind: "max"; value: number; message?: string }
+  | { kind: "min"; value: number; message?: string }
+  | { kind: "type"; value: string | string[]; message?: string };
+
+export interface ZodFileDef extends ZodTypeDef {
+  typeName: ZodFirstPartyTypeKind.ZodFile;
+  checks: ZodFileCheck[];
+}
+
+export class ZodFile<FileConstructor extends Function> extends ZodType<
+  FileConstructor,
+  ZodFileDef
+> {
+  constructor(def: ZodFileDef, private readonly fileConstructor: Function) {
+    super(def);
```

Comment:
Can you add here the `description` field? Usually this is not used. But they way I'm trying to use it is helpful to be able to see if current validator is a `ZodFile`

``` javascript
constructor(def: ZodFileDef, private readonly fileConstructor: Function) {
  super(def);
  def.description = "ZodFile";
}
```

Reply:
Ideally this should be `typeName` but i think the getter is not exposed in the base class `ZodType` right?